docs/rfc/rfc-7.adoc

```diff
@@ -0,0 +1,304 @@
+clients in the group a chance to properly recognize a Coordinator rotation.
+
+If we sweep early because a lot of deposits have been revealed, then we don't
+"push back" our scheduled `sweep_period` sweep. Rather, that sweep continues
+as planned.
+
+*Example*: We have a `sweep_period = 8 hours` and `sweep_max_deposits = 15`. At
+08:00 UTC, a sweep just occurred, and the next is scheduled for 16:00 UTC.
+At 10:00 UTC, 15 deposits get revealed which triggers an early sweep due to
+`sweep_max_deposits`. Rather than pushing back the next scheduled sweep to
+18:00 UTC, it remains at 16:00 UTC.
+
+==== Sweep transaction
+
+The Coordinator chooses sufficiently confirmed deposits for the sweep. To avoid
+synchronization problems with deposits on the brink of being sufficiently
+confirmed, the Coordinator may choose deposits with more than 6 confirmations
+on Bitcoin and 20 confirmations on Ethereum. To avoid problems with proving
+large transactions and too many signatures to be executed, the Coordinator
+should not pick more than 15 deposits. The Coordinator chooses the Bitcoin
+transaction fee reflecting the current situation in the Bitcoin network and the
+maximum possible fee accepted by the Bridge. To leave enough time for retries,
+deposits for which Bitcoin transactions unlock for a refund earlier than two
+weeks from now are skipped (`sweeping_refund_safety_time`).
+
+The Coordinator must not propose a sweep if there is currently a pending signing
+operation being executed by the wallet or if the last transaction from the
+wallet does not have 6 Bitcoin confirmations yet.
+
+The Coordinator proposes a sweep over the wallet's broadcast channel.
+
+All signing group members must confirm that all deposits in the proposed sweep
+are sufficiently confirmed, that the maximum number of deposits is not exceeded,
+that the proposed fee reflects the current situation in the Bitcoin network
+plus/minus some margin, that the fee does not exceed the maximum one allowed
+by the Bridge, that all deposit UTXOs can be unlocked with the wallet's public
+key, and that there is enough time to unlock UTXO before the refund.
+
+When proposing a sweep, the Coordinator sets the Ethereum block number at which
+the signing protocol should start. All clients validate if that block's number
+is +-2 blocks from the current one.
+
+If all these requirements are met, the last transaction executed by the wallet
+is sufficiently confirmed, and there is no other signing by the wallet in
+progress, the signing group members proceed with tECDSA signing protocol.
+
+Each UTXO being an input to the sweep transaction is unlocked sequentially in
+a separate signing session. Each signing session begins with an announcement
+phase allowing to exclude offline operators. There is a 30-blocks (~6 minutes)
+timeout for each signing attempt.
+
+If the given signing attempt fails for any reason (error or timeout), the next
+attempt starts exactly 30 blocks after the previous one started.
```

Comment:
I think this is not "exactly" as we have the announcement phase and a small cooldown period.